python/plugins/processing/gui/AlgorithmDialog.py

```diff
@@ -154,7 +155,19 @@ def getParameterValues(self):
                     if self.mainWidget().checkBoxes[param.name()].isChecked():
                         dest_project = QgsProject.instance()
 
-                value = self.mainWidget().outputWidgets[param.name()].getValue()
+                widget = self.mainWidget().outputWidgets[param.name()]
+                value = widget.getValue()
+                outputPath, _ = value.sink.valueAsString(self.context.expressionContext())
+                if outputPath.split(":")[0] not in ["memory", "postgis", "ogr"]:
+                    ext = os.path.splitext(outputPath)[1][1:]
+                    supportedExtensions = []
+                    if isinstance(param, QgsProcessingParameterRasterDestination):
```

Comment:
Do we need to consider case insensitivity here?

Reply:
fixed